src/main/java/com/applause/auto/testrail/client/TestRailClient.java

```diff
@@ -522,4 +535,18 @@ public List<TestDto> getTestResultsForRun(
   private void debugLogPostResponse(final int httpStatusCode, @NonNull final String methodName) {
     log.debug("Retrieved HTTP " + httpStatusCode + " from TestRail " + methodName + " request.");
   }
+
+  private <T> Response<T> makeCall(
+      Function<TestRailApi, CompletableFuture<Response<T>>> testRailAction)
+      throws TestRailException {
+    try {
+      return testRailAction.apply(apiClient).join();
+    } catch (CompletionException e) {
+      if (e.getCause() instanceof SocketTimeoutException) {
+        throw new TestRailException(TestRailErrorStatus.SOCKET_TIMEOUT);
+      } else {
+        throw new TestRailException(TestRailErrorStatus.UNKNOWN_ERROR);
```

Comment:
include extra info here so we can potentially identify/resolve these later.

Reply:
These exceptions can be shown in admin panel (and other places) in some cases. I might add a log statement for the UNKNOWN_ERROR case at a lower log level

Reply:
I'm more concerned with prod troubleshooting, specifically the unknown-error.
I'm good with DEBUG log since we can set class-level logging in UI for timeperiod.